### ISSUE TITLE
Nullable ref properties behave inconsistently with byref arguments

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#40889]
**Version Used**: 
```
commit a0404639df1ab23612390052d0b5d400c864bc38
Merge: eb26598979 6aa3e57ac0
Author: Sam Harwell <sam.harwell@microsoft.com>
Date:   Mon Dec 30 20:03:18 2019 -0800

    Merge pull request #40643 from sharwell/driver-tasks

    Reduce Task allocations in the analyzer driver
```

**Steps to Reproduce**:

Compile and run the following code
```cs
class C
{
  string? field; 
  [NotNull] public ref string? Prop
  {
    get
    {
      field = """";
      return ref field; 
    }
  }
  
  void M()
  {
    Prop.ToString();
    Prop = null;
    Prop.ToString(); // no warnings
    Out(out Prop);
    Prop.ToString(); // no warnings
    Ref(ref Prop);
    Prop.ToString(); // CS8602
  }
  
  public void Ref(ref string? s) => s = null;
  public void Out(out string? s) => s = null;
}
```

**Expected Behavior**:
No warnings.

**Actual Behavior**:
`CS8602: Dereference of a possibly null reference.` is reported only for the last derefernece of `Prop`.

**Notes**
The allowed inputs for the property `Prop` include null values and allowed outputs are always not null according to its annotation.
Note that actual writes via `= null` and `out Prop` respect this and don't update the `Prop` state to nullable. However `ref` arguments don't adjust the set value based on the passed members' annotations
